docs/mediu/apcm.md

```diff
@@ -187,11 +187,145 @@ int main() {
 }
 ```
 
+## Algoritmul lui Boruvka
+
+!!! info "Definiție" 
+    Acest algoritm începe cu fiecare nod fiind într-o comopnentă conexă doar cu el însuși. Apoi, va face iterații prin graf, până când nu este arbore (adică cât timp mai sunt cel puțin două componente conexe). El va găsi pentru fiecare componentă conexă (sau pentru fiecare nod, depinzând de problemă) cea mai bună muchie nefolosită care o (îl) unește de altă componentă conexă. După ce aceste muchii sunt găsite, ele sunt folosite. Vom folosi și la acest algoritm structura Union-Find pentru a afla dacă muchiile duc la componente conexe diferite și pentru a uni două componente conexe.
+
+!!! note "Observație"
+    La fiecare iterare prin graf, numărul de componente conexe se înjumătățește. La început sunt $n - 1$ componente conexe, deci se vor face $O(\log n)$ iterații.
+
+Aici puteți găsi o implementare în C++ a algoritmului lui Boruvka:
```

Comment:
As pune o observatie pentru cazul cand ai raspunsul impossible (eu unul am facut asta cu un bool in care verific daca am adaugat o muchie in pasul curent, dar ma gandesc ca pot fi si alte variante, poate sa verifici daca graful e conex daca tot ai DSU deja?)

Reply:
https://cses.fi/paste/2a09f78601388a1fa5047b/